docs/adr/0007-modelling-history.md

```diff
@@ -0,0 +1,272 @@
+  and URL so this can be presented to users;
+- Ability to re-present either the live or draft version to Publishing API
+  should there be any problems with publishing.
+
+And with regards to history:
+
+- To know who did what to a document and when - effectively an audit trail of
+  accountability;
+- Be able to compare two revisions of a document to show changes;
+- Have stored in our database a full history of a document, which may go back
+  further than the Publishing API (Whitehall has longer histories of some
+  published documents)
+
+A [MoSCoW](https://en.wikipedia.org/wiki/MoSCoW_method) breakdown of this
+functionality was defined as:
+
+### Must
+
+- Store full information on latest and live versions of a document
+- Consider how a revision of a document and aspects of it have state that may
+  change
+
+### Should
+
+- Have the full information for every edition that was published on GOV.UK
+- Record every change a publisher makes and have that information accessible
+- Consider states to be something that may change in a background job
+- Avoid relying on the TimelineEntry model as the sole answer to any historical
+  queries
+
+### Could
+
+- Link TimelineEntry to changes rather than storing the information twice
+- Enable a future feature to enable reverting to different points in time
+- Separate related data to different tables to reduce data usage
+
+### Won’t
+
+- Consider the sharing of data between translations, as per
+  [ADR-3](../adr/0003-initial-domain-modelling.md#decision)
+
+## Decision
+
+The proposed data model expands on the definitions of entities accordingly:
+
+**Document**: A record that represents all versions of a piece of content in a
+particular locale. It has many editions and many revisions. At any time it will
+have a current edition -  shown on Content Publisher index - and potentially a
+live edition which is currently on GOV.UK. The live and current edition can be
+the same. Document serves as a mutable entity that tracks when a Document was
+last edited and it is expected to be a joining point for document related data
+that is not associated with a particular editions.
+
+**Edition**: An edition is a numbered version of a document that has been, or is
+expected to be, published on GOV.UK. It is associated with a current revision
+and maintains an association with all revisions that have been associated with
+it. It has a current state (e.g. draft, submitted, published)
+and has knowledge of the past states it held. It is mutable and is expected
+as a place where any edition level database constraints can be placed.
+
+**Revision**: Represents the content of a document at a particular point in
+time. It has a number to indicate which revision of the document it is and
+stores who created it. Any time a user changes an aspect of a document a new
+revision is created and the current edition is updated to reference that
+revision - a revision cannot be edited and is considered immutable. A revision
```

Comment:
Just to clarify, a revision refers to one set of changes? So if a user decides they want to make a change to a news article, they create an edition for each state change that the news article goes through (draft, submitted for review, published) etc, but these all refer to the same revision?

Reply:
That's correct 👍